### PR TITLE
Feature: Add `HitTestBehavior` as an argument to `Dismissible`

### LINKED ISSUE #62270: Make Dismissible's HitTestBehavior an argument
Dismissible currently has the [HitTestBehavior hard-coded as opaque](https://github.com/flutter/flutter/blob/1ad9baa8b9/packages/flutter/lib/src/widgets/dismissible.dart#L580). That makes some (admittedly edge-case) layouts impossible to do without a lot of copy/paste code. There doesn't seem to be a reason why the HitTestBehavior couldn't be an argument for Dismissible.

My example: putting a horizontally dismissible widget in a Stack doesn't let me vertically scroll the widgets underneath, but hacking in HitTestBehavior.translucent, it works great.

### PR DESCRIPTION
**Pull request by Inconnu08**
## Description

Allow `Dismissible` to have `HitTestBehavior` property.

## Related Issues

Fixes #62270 

## Tests

I added the  following test:
`HitTestBehaviour defaults to opaque but also respects the specified behavior`

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/dismissible.dart [hunk 1]
```diff
@@ -187,6 +189,9 @@ class Dismissible extends StatefulWidget {
   ///  * [DragGestureRecognizer.dragStartBehavior], which gives an example for the different behaviors.
   final DragStartBehavior dragStartBehavior;
 
+  /// How to behave during hit tests.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/dismissible.dart:192]:
```suggestion
  /// How to behave during hit tests.
  ///
  /// This defaults to [HitTestBehavior.opaque].
```

**Review by Piinks — COMMENTED** [review]
Hi @Inconnu08 can you add a test to verify the expected behavior? This also prevents the change from being undone in a future change. :)

**Review by Piinks — COMMENTED** [review]
It looks like the latest update here is causing a lot of tests to fail. Can you take a look @Inconnu08 ?

**Comment by Inconnu08** [comment]
@Piinks  it looks like a lot of tests failing aren't even related to my changes. 😟

**Comment by Piinks** [comment]
> @Piinks it looks like a lot of tests failing aren't even related to my changes. 😟

Ok, it could be that you need to update you branch then. Can you update with the latest from master?

File: packages/flutter/test/widgets/dismissible_test.dart [hunk 2]
```diff
@@ -735,6 +735,38 @@ void main() {
...
+            key: third,
+            child: Text('text'),
+          ),
+        ],
+      ),
+    ));
+
+    expect((tester.firstWidget(find.byKey(first)) as Dismissible).hitTestBehavior, HitTestBehavior.opaque);
+    expect((tester.firstWidget(find.byKey(second)) as Dismissible).hitTestBehavior, HitTestBehavior.translucent);
+    expect((tester.firstWidget(find.byKey(third)) as Dismissible).hitTestBehavior, HitTestBehavior.deferToChild);
```
Comment (goderbauer) [inline packages/flutter/test/widgets/dismissible_test.dart:767]: This is not really testing that the `Dismissible` actually honors the setting.

**Comment by goderbauer** [comment]
Looks like there's another PR out that does the same thing: https://github.com/flutter/flutter/pull/64379

**Comment by Piinks** [comment]
I am going to close this since it looks like #64379 is just about done and implements the same thing. Thanks for the contribution!

**Pull request closed without merging**